### ISSUE TITLE
Crash in [Talkable offerTargetFor:]

### ISSUE THREAD

**Issue by ehyche** [talkable/ios-sdk#20]
We are seeing a crash in Crashlytics that looks like this:

<img width="1308" alt="Screenshot 2023-01-26 at 2 59 58 PM" src="https://user-images.githubusercontent.com/4141428/214937871-f4f82edd-5e41-438c-ac4e-1dd497f3591d.png">

The code in question is here:

```
- (TKBLOfferTarget*)offerTargetFor:(WKWebView*) webView {
    if (!_offerTargets) {
        _offerTargets = [NSMutableArray new];
    }
    for (TKBLOfferTarget* target in _offerTargets) {
        if (![target isUsed])
            [_offerTargets removeObject:target];
    }
    TKBLOfferTarget* target = [[TKBLOfferTarget alloc] initWithWebView:webView];
    [_offerTargets addObject:target];
    return target;
}
```

and it does indeed look like if `[target isUsed]` returns `NO`, then we will mutate the `_offerTargets` array while we are iterating it.

**Comment by dgtv** [comment]
Hey @ehyche,
Thanks for the point and clear explanation.
Looks like there is indeed a bug in the SDK, but this is the first time we have it noticed.
We can arrange the fix to be made to the SDK in 3-4 week due to our development capacity and processes. Will it work for you?

**Comment by ehyche** [comment]
@dgtv : Thanks for the response! Sure, that will work. Thank you.

**Comment by ehyche** [comment]
@dgtv : any update on this?

**Comment by dgtv** [comment]
@ehyche Yes, thanks for the reminder. We already released the new version of SDK 1.4.15 yesterday with the bugfix and it is available from all distribution places listed in the documentation: https://docs.talkable.com/ios_sdk/getting_started.html.
Thanks for your contribution, we appreciate it.
